File: src/python/PythonSDK/foundationallm/models/orchestration/completion_request.py
from typing import List, Optional
from pydantic import Field
from foundationallm.models.metadata import Agent
from foundationallm.models.metadata import DataSource
from foundationallm.models.language_models import LanguageModel, EmbeddingModel
from .completion_request_base import CompletionRequestBase

class CompletionRequest(CompletionRequestBase):
    """
    Orchestration completion request.
    """
    type: Optional[str] = Field(None, alias='$type')
    agent: Optional[Agent] = None
    data_sources: Optional[List[DataSource]] = []
    language_model: Optional[LanguageModel] = None
    embedding_model: Optional[EmbeddingModel] = None    

File: src/dotnet/Common/Models/Orchestration/CompletionRequest.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Orchestration;

/// <summary>
/// The completion request object.
/// </summary>
public class CompletionRequest : CompletionRequestBase
{
    /// <summary>
    /// The name of the selected agent.
    /// </summary>
    [JsonPropertyName("agent_name")]
    public string? AgentName { get; set; }


    /// <summary>
    /// A list of Gatekeeper feature names used by the orchestration request.
    /// </summary>
    [JsonPropertyName("gatekeeper_options")]
    public string[]? GatekeeperOptions { get; set; }

    /// <summary>
    /// Settings that override some aspects of behaviour of the orchestration.
    /// </summary>
    [JsonPropertyName("settings")]
    public OrchestrationSettings? Settings { get; set; }
}
